WalletWasabi/WabiSabi/Models/DecompositionAlgs/RestrictedSetDecomposer.cs

```diff
@@ -0,0 +1,56 @@
+using NBitcoin;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class RestrictedSetDecomposer
+	{
+		private Decomposition Decomposition { get; } = new();
+
+		public RestrictedSetDecomposer()
+		{
+			Decomposition.Extend(new Coin() { Amount = Money.Zero });
+		}
+
+		public virtual Money? PickNextAmount(Money remaining, Money dustThreshold)
```

Comment:
`dustThreshold` looks like a candidate for a property/field.

Reply:
yes